eng/targets/CSharp.Common.targets

```
@@ -36,7 +36,7 @@
 
     <!-- Enable .NET code style analysis during build for src projects. -->
     <!-- Workaround bug where turning this on produces warnings in VS -->
-    <EnforceCodeStyleInBuild Condition="'$(VisualStudioVersion)' &lt; '17.0'">false</EnforceCodeStyleInBuild>
+    <EnforceCodeStyleInBuild Condition="'$(BuildingInsideVisualStudio)' == 'true' AND '$(VisualStudioVersion)' &lt; '17.0'">false</EnforceCodeStyleInBuild>
```

same nit:
```suggestion
    <EnforceCodeStyleInBuild Condition=" '$(MSBuildRuntimeType)' != 'Core' AND '$(VisualStudioVersion)' &lt; '17.0' ">false</EnforceCodeStyleInBuild>
```

Separately have we confirmed `<` works with strings that look like floating-point numbers❔ @rainersigwald❔

---

it does but consider using the newish [version comparison property functions](https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2019#msbuild-version-comparison-functions)